### ISSUE TITLE
yum module fails to update package

### ISSUE THREAD

**Issue by aabdnn** [ansible/ansible#22902]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
yum module

##### ANSIBLE VERSION
```
ansible 2.2.1.0
  config file = 
  configured module search path = Default w/o overrides
```

##### CONFIGURATION
ANSIBLE_HOST_KEY_CHECKING=no
ANSIBLE_SSH_PIPELINING=yes
ANSIBLE_INVENTORY=/Users/anandb/repo/ansible/hosts
ANSIBLE_GATHERING=smart
ANSIBLE_REMOTE_TEMP=/tmp
ANSIBLE_SSH_CONTROL_PATH=/tmp/%%r@%%h
ANSIBLE_HASH_BEHAVIOUR=merge
ANSIBLE_TIMEOUT=75
ANSIBLE_RETRY_FILES_ENABLED=no

##### OS / ENVIRONMENT
Managing from MacOX 10.12.3. The managed host is CentOS 6

##### SUMMARY
Asking the yum module to update a package to the latest version doesn't achieve any result.

##### STEPS TO REPRODUCE
```
ansible host -bK -m yum -a 'name=package state=latest update_cache=yes'
```

##### EXPECTED RESULTS
The package should have been updated.

##### ACTUAL RESULTS
The package is not updated. I know there is a new version available, because on the host I see:
```
# yum check-update
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: mirror.proserve.nl
 * epel: mirror.1000mbps.com
 * extras: mirror.previder.nl
 * gii: cf3-ext.ripe.net
 * updates: mirror.proserve.nl

atlasprobe.x86_64
```

```
ansible <host> -bK -m yum -a 'name=atlasprobe state=latest update_cache=yes' -vvvv
No config file found; using defaults
SUDO password: 
Loading callback plugin minimal of type stdout, v2.0 from /usr/local/lib/python2.7/site-packages/ansible/plugins/callback/__init__.pyc
Using module file /usr/local/lib/python2.7/site-packages/ansible/modules/core/packaging/os/yum.py
<host> ESTABLISH SSH CONNECTION FOR USER: None
<host> SSH: EXEC ssh -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=75 -o ControlPath=/tmp/%r@%h host '/bin/sh -c '"'"'sudo -H -S  -p "[sudo via ansible, key=nistixrusxmctqyoehpfdamjjerzdkqv] password: " -u root /bin/sh -c '"'"'"'"'"'"'"'"'echo BECOME-SUCCESS-nistixrusxmctqyoehpfdamjjerzdkqv; /usr/bin/python'"'"'"'"'"'"'"'"' && sleep 0'"'"''
host | SUCCESS => {
    "changed": false, 
    "invocation": {
        "module_args": {
            "conf_file": null, 
            "disable_gpg_check": false, 
            "disablerepo": null, 
            "enablerepo": null, 
            "exclude": null, 
            "install_repoquery": true, 
            "list": null, 
            "name": [
                "atlasprobe"
            ], 
            "state": "latest", 
            "update_cache": true, 
            "validate_certs": true
        }, 
        "module_name": "yum"
    }, 
    "msg": "", 
    "rc": 0, 
    "results": [
        "All packages providing atlasprobe are up to date", 
        ""
    ]
}
```

**Comment by krzysztof-magosa** [comment]
Could you run `yum check-update` and show its exit code (`echo $?`)?

**Comment by aabdnn** [comment]
# yum check-update
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: mirrors.noction.com
 * epel: epel.mirror.nucleus.be
 * extras: mirrors.noction.com
 * gii: cf3-ext.ripe.net
 * updates: mirrors.noction.com

atlasprobe.x86_64                                                                   4770-2                                                                   gii
[root@nl-ams-as3333-preprod ~]# echo $?
100

**Comment by t794104** [comment]
I have the same problem; Yum module not updating a package already installed with state = latest.

After some debugging it found out that the 'check-update' command is running from the function 'latest' and that the output is modified with a Regex. 

out variable before regex : 
"Loaded plugins: langpacks, product-id, search-disabled-repos, subscription-\n              : manager\nThis system is not registered to Red Hat Subscription Management. You can use subscription-manager to register.\n\nnems-it-ontw-devops.x86_64             0.0.4-1.rhel7             nems-it_generic\n"

out variable after regex :
"Loaded plugins: langpacks, product-id, search-disabled-repos, subscription- manager\n**This system is not registered to Red Hat Subscription Management. You can use subscription-manager to register. nems-it-ontw-devops.x86_64             0.0.4-1.rhel7             nems-it_generic**\n",

By removing the '\n' the first package with a update get's concated to the previous line (bold).... and ignored.

By changing the code from

```
    # run check-update to see if we have packages pending
    rc, out, err = module.run_command(yum_basecmd + ['check-update'])
```
to 
```
    # run check-update to see if we have packages pending
    rc, out, err = module.run_command(yum_basecmd + ['check-update', '-q'])
```
The message line's aren't printed to the output and the regex (which is there for a reason... i think) work as it should.

**Comment by alikins** [comment]
This sounds like a dupe of either https://github.com/ansible/ansible/issues/20608 (fixed in 2.3.0) and backported to stable-2.2 in e28cea7e0520f8d1b81e57a9951d55a4feb3e34b

Or possibly a dupe of https://github.com/ansible/ansible/issues/24330  (fixed in devel and backported to stable-2.3)
